testsuite/tests/misc/sort_segment.ml

```diff
@@ -0,0 +1,47 @@
+(* TEST *)
+
+open Printf
+
+(* [generate n] generates an array of length [n],
+   containing random integer data. *)
+let generate n =
+  Array.init n @@ fun i -> Random.int n
```

Comment:
If you use the global RNG, I think it would be better if you could initialize it explicitly: either you call `Random.self_init ()`, if you intend the test to be non-deteministic, or you call `Random.init 42` (or you favorite seed), if you intend the test to be deterministic. Right now it looks like you forgot to make a decision about it.

Reply:
Thanks. I have added `Random.init 42`.